contexts/README.md

```diff
@@ -1,6 +1,27 @@
 # DID Contexts
 
-## Background
+## Terms
+
+The terms present in the DID Core context (v1) are normatively defined in
+the [DID Core Specification](https://www.w3.org/TR/did-core), except for
+the following terms, which are defined in [the Security Vocabulary](w3id.org/security/)
+and maintained here for developer convenience.
+
+* `publicKey`
+* `blockchainAccountId`
+* `EcdsaSecp256k1VerificationKey2019`
+* `Ed25519Signature2018`
+* `Ed25519VerificationKey2018`
+* `JsonWebKey2020`
+* `JsonWebSignature2020`
+* `RsaVerificationKey2018`
+* `SchnorrSecp256k1VerificationKey2019`
+* `X25519KeyAgreementKey2019`
```

Comment:
We've been discussing this internally at Digital Bazaar today... @dlongley is concerned about this approach and suggest something different... and that is that each cryptosuite should get its own context... so, something akin to this would go in a DID Document:

```
"@context": [
  "https://www.w3.org/ns/did/v1", 
  "https://www.w3.org/2020/ed25519/v1", 
  "https://www.w3.org/2020/jwk/v1", 
  "https://www.w3.org/2021/secp256r1/v1"
]
```

The downside is more contexts (but only the ones your DID Method supports). The upside is that each context is simple, self-contained, and the churn we're going through right now with the vc-data-model spec ends up not becoming a thing in the future. That is, it would provide more stability to the VC and DID Contexts to define the cryptosuites outside of them. This is akin to what developers do today with #include files and import statements. You just pull in the stuff you need, where each file is meant to do something fairly specific.

I'll try to write this up as a more concrete proposal and send it to the W3C CCG mailing list (with DID WG cc'd).

/cc @tplooker @kdenhartog @rhiaro @OR13 @dmitrizagidulin @dlongley @davidlehn

Reply:
+1 agreed, this is better

Reply:
```
"@context": [
  "https://www.w3.org/ns/did/v1", 
  "https://www.w3.org/2020/ed25519/v1", 
  "https://www.w3.org/2020/jwk/v1", 
  "https://www.w3.org/2021/secp256r1/v1"
]
```

1. I hate the date stamping
2. I hate the https://www.w3.org root, cant't maintain things, feels like a black whole were a council of elders will block us from basic changes
3. Encourages incompatibility (no assumed safe defaults, like ed25519)
4. Almost no point in did/v1... from a practical standpoint, what is it doing?


Instead I would prefer to see:

https://www.w3.org/ns/did/v1 - > self contained minimal set of crypto and terms.
https://www.w3.org/ns/did/v2 -> self contained minimal set of crypto and terms with breaking changes (dropping support for legacy crypto, for example).

I don't have anything against "https://www.w3.org/2020/ed25519/v1"... as a thing that might exist somewhere... its very useful to have contexts that only define terms in a limited way... for testing... from a usability standpoint, it sucks...

as a developer working on a standard, I want to pick a version and have it work... its our job to make the version contain the right stuff for the right moment in time, and to release new versions when recommendations / best practices change.

Reply:
> I hate the https://www.w3.org root, cant't maintain things, feels like a black whole were a council of elders will block us from basic changes

once we lay claim to a w3.org URI we can have them point it at github pages (or whatever) and update it as much as we want. w3.org has a persistence guarantee, so it's good to use those URIs when we have an option to (as a WG or CG). If we pointed at a different _server_, we'd lose the w3c server backups as a feature, but that's a tradeoff with speed of changes being made live. If we hosted them directly on the w3c servers, we just need a staff contact to make the upload - as long as who is responsible for that URI is (eg. a CG) is clear, and that group has consensus on changes, w3c staff are not going to block anything.

Reply:
as soon as its proved that the w3.org root redirects to github pages / work as easily as security-vocab, I would withdraw my objection to that part, assuming I can see the CODEOWNERs for the repo, and know who to tar an feather for holding up merges :)

Reply:
I tend to like separating them out and am +.3 to doing them all separately, but curious to hear the reasoning behind doing it all separately rather than combining them together in the security context. The one downside I see to splitting them out further is that when protected terms are being redefined inside a suite such as the `proofPurpose` issue we had with v3 context then it would be much harder to identify the values. However, it also would drop the number of terms that we should actually be defining in the base v3 context and instead would allow them to be defined as a scoped term within a suite.

Reply:
> as soon as its proved that the w3.org root redirects to github pages / work as easily as security-vocab, I would withdraw my objection to that part, assuming I can see the CODEOWNERs for the repo, and know who to tar and feather for holding up merges :)

https://www.w3.org/2019/did-wg/ is the home page for this Working Group, and its content is on GitHub because it is redirected to the group's repo. I can give you other examples. 

To be precise, though: what can be easily redirected are either files under https://www.w3.org/ns or https://www.w3.org/2020/ (or 2021, or 2022). Directories right under https://www.w3.org are considered to be special real estate.

All this to say: using the URI-s proposed by @msporny and be redirected to a file on GitHub is perfectly possible.

---

Putting my staff contact's job aside for a moment: I do like the idea of separate context files. It makes things much clearer and cleaner, at least for me.